Add ability to define DLRN driver in INI config file

jcapiitao opened the pull request:
We set 'downstream_driver' as default value for 'driver' key.

jpichon commented on patch_rebaser/patch_rebaser.py:
```diff
@@ -426,6 +427,7 @@ def main():
 
     # Default values for options in patch_rebaser.ini
     defaults = {
+        'driver': 'downstream_driver',
```
If this is an addition to patch_rebaser.ini, could you also update [patch_rebaser.ini.example](https://github.com/release-depot/patch_rebaser/blob/master/patch_rebaser/patch_rebaser.ini.example) with the new configuration option + default, and a short explanation?

jpichon reviewed (changes requested):
Thank you for the patch. Could you clarify a little bit the background for the change, maybe in the commit message?

Additionally, it would be good to add a unit test or two to show that the new driver value gets picked up where expected, and that the changes are transparent/backwards-compatible otherwise. Some of the unit tests data fixtures need to be updated with the new config value, although it would be good to also have a test where it's still missing anyway to prove that the patch_rebaser.py will be backward compatible. Thanks!

(I think some of the Python versions that failed with "Cancelled" are no longer maintained, I'll see about cleaning up the test matrix, although if you happen to have a few spare cycles to do it in a separate PR (updating tox, github actions should cover it I think), that would be super helpful!)

jcapiitao commented:
> Thank you for the patch. Could you clarify a little bit the background for the change, maybe in the commit message?
> 
> Additionally, it would be good to add a unit test or two to show that the new driver value gets picked up where expected, and that the changes are transparent/backwards-compatible otherwise. Some of the unit tests data fixtures need to be updated with the new config value, although it would be good to also have a test where it's still missing anyway to prove that the patch_rebaser.py will be backward compatible. Thanks!
> 
> (I think some of the Python versions that failed with "Cancelled" are no longer maintained, I'll see about cleaning up the test matrix, although if you happen to have a few spare cycles to do it in a separate PR (updating tox, github actions should cover it I think), that would be super helpful!)

I updated the check jobs in first commit.
I also updated the unit tests to take into account your comments.

jpichon reviewed (changes requested):
Sorry for the delay and thank you for the updated commit message. This is really helpful and now I understand better what we are trying to do (so hopefully reviews won't take as long!)

This needs a rebase, as someone fixed the CI issues in all of our repos so the .github/workflows stuff was fixed separately.

I would like to see a couple of new unit tests proving the new behaviour and to help us avoid regressions in the future. I made a couple of suggestions inline and hope the examples are helpful!
